integration/test/watcher/configmap/configmap_test.go

```diff
@@ -0,0 +1,355 @@
+			},
+			Spec: v1alpha1.AppSpec{
+				Catalog: key.DefaultCatalogName(),
+				KubeConfig: v1alpha1.AppSpecKubeConfig{
+					InCluster: true,
+				},
+				Name:      key.TestAppReleaseName(),
+				Namespace: key.Namespace(),
+				UserConfig: v1alpha1.AppSpecUserConfig{
+					ConfigMap: v1alpha1.AppSpecUserConfigConfigMap{
+						Name:      key.UserConfigMapName(),
+						Namespace: key.Namespace(),
+					},
+				},
+				Version: "0.1.0",
+			},
+		}
+
+		_, err = config.K8sClients.G8sClient().ApplicationV1alpha1().Apps(key.Namespace()).Create(ctx, appCR, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating %#q app cr", key.TestAppReleaseName()))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting until user configmap being labelled")
+
+		o := func() error {
+			cm, err := config.K8sClients.K8sClient().CoreV1().ConfigMaps(key.Namespace()).Get(ctx, key.UserConfigMapName(), metav1.GetOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if _, ok := cm.GetLabels()[pkglabel.Watching]; !ok {
+				return microerror.Maskf(notFoundError, fmt.Sprintf("%#q label not found", pkglabel.Watching))
+			}
+
+			return nil
+		}
+
+		n := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to get a label; retrying in %d", t), "stack", fmt.Sprintf("%v", err))
+		}
+
+		b := backoff.NewMaxRetries(5, backoff.ShortMaxInterval)
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waited until user configmap being labelled")
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting until appcatalog configmap being labelled")
+
+		o := func() error {
+			cm, err := config.K8sClients.K8sClient().CoreV1().ConfigMaps(key.Namespace()).Get(ctx, key.AppCatalogConfigMapName(), metav1.GetOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if _, ok := cm.GetLabels()[pkglabel.Watching]; !ok {
+				return microerror.Maskf(notFoundError, fmt.Sprintf("%#q label not found", pkglabel.Watching))
+			}
+
+			return nil
+		}
+
+		n := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to get a label; retrying in %d", t), "stack", fmt.Sprintf("%v", err))
+		}
+
+		b := backoff.NewMaxRetries(5, backoff.ShortMaxInterval)
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waited until appcatalog configmap being labelled")
+	}
+
+	var updatedResourceVersion string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("editing configmap %#q in namespace %#q", key.UserConfigMapName(), key.Namespace()))
```

Comment:
```suggestion
		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating values in configmap %#q in namespace %#q", key.UserConfigMapName(), key.Namespace()))
```

The user will edit the values but it reads a bit off for me. WDYT to this?

Reply:
Nicer than editing I think, thanks for a suggestion!